rllib/evaluation/rollout_worker.py

```diff
@@ -640,15 +642,30 @@ def wrap(env):
         # TODO(jungong) : clean up after non-connector env_runner is fully deprecated.
         self.filters: Dict[PolicyID, Filter] = {}
         for (policy_id, policy) in self.policy_map.items():
-            filter_shape = tree.map_structure(
-                lambda s: (
-                    None
-                    if isinstance(s, (Discrete, MultiDiscrete))  # noqa
-                    else np.array(s.shape)
-                ),
-                policy.observation_space_struct,
-            )
-            self.filters[policy_id] = get_filter(self.observation_filter, filter_shape)
+            if policy_config.get("enable_connectors"):
+                ctx = ConnectorContext.from_policy(policy)
+                connector = get_synced_filter_connector(
```

Comment:
do we really need to do this here?
can we just call get_synced_filter_connector() in get_agent_connectors_from_config()?
https://github.com/ray-project/ray/blob/master/rllib/connectors/util.py#L25

Reply:
any input to this question I had?
just wondering if we can keep things simpler and not bring connectors up to the rollout worker level.

Reply:
In the end it's definitly cleaner to move this. But right now this code block still depends on `self.observation_filter` which is ugly to remove or sneak into `get_agent_connectors_from_config()`. I think we should leave this here for the moment. When connectors are switched on by default, we can refactor a little bit. This whole PR is designed in the way that the connectors solution and the old solution share lots of code and I would like to do another PR that not only remove the other solution, but also makes the connectors solution more elegant.

Reply:
ok get it.
I am ok with accessing policy.agent_connectors here to find the filter agent, and save a reference to its filter object. as long as we add some comments explaining it :)
but can we construct the filter agent in get_agent_connectors_from_config()? there is no reason to create the filter agent here and append it right?
we can also add a ```get(self, name: str)``` API to ```ConnectorPipeline``` so it's easier to check/find a specific connector.

the reason I am hoping to create all the connectors in a centralized place is that we actually print the complete connector setup in get_agent_connectors_from_config(). if we are gonna add things somewhere else, this message is not correct anymore.

Reply:
I understand. I've implemented this. Let's see if tests pass, I might have to tinker a little more.

Reply:
Moved this as you requested!